Source/Global/mem.h

```diff
@@ -75,6 +90,21 @@ class http_stl_allocator
     }
 };
 
+template<typename T>
+std::shared_ptr<T> http_allocate_shared()
+{
+    return std::allocate_shared<T, http_stl_allocator<T>>(http_stl_allocator<T>());
+}
+
+template<typename T>
+std::unique_ptr<T, http_alloc_deleter<http_stl_allocator<T>>> http_allocate_unique()
```

Comment:
```c++
template<typename T>
using HC_UNIQUE_PTR = std::unique_ptr<T, http_alloc_deleter<http_stl_allocator<T>>>;

template<typename T, typename... Args>
HC_UNIQUE_PTR<T> http_allocate_shared(Args&&... args)
{
    http_stl_allocator<T> alloc;
    auto p = std::allocator_traits<http_stl_allocator<T>>::allocate(alloc, 1);
    auto o = new(p) T(std::forward<Args>(args)...);
    return HC_UNIQUE_PTR<T>(o);
}
```

1. Made into a variadic template to allow passing arguments through to the constructor
2. Added template alias for std::unique_ptr with custom deleter
3. Used the pointer returned by operator new instead of the one returned by the allocator (they are probably identical but...)
4. Written in the browser so it may fail to compile

Reply:
done